cave/utils/csv2rh.py

```diff
@@ -0,0 +1,250 @@
+            parameters = [p[2:] for p in self.data.columns if p.startswith('p_')]
+            with open(self.pcs, 'r') as fh:
+                cs = pcs.read(fh)
+            diff = set(cs.get_hyperparameter_names()).symmetric_difference(set(parameters))
+            if not len(diff) == 0:
+                raise ValueError("Comparing parameters from \"%s\" and csv. "
+                                 "In pcs but not in csv: [%s]. "
+                                 "In csv but not in pcs: [%s]. " % (self.pcs,
+                                 str(set(cs.get_hyperparameter_names()).difference(set(parameters))),
+                                 str(set(parameters).difference(set(cs.get_hyperparameter_names())))))
+        else:
+            parameters = [p for p in self.data.columns if p.startswith('p_')]
+            self.logger.warning("No parameter configuration space (pcs) provided! "
+                                "Interpreting all parameters as floats. This might lead "
+                                "to suboptimal analysis.")
+            minima = self.data.min()  # to define ranges of hyperparameters
+            maxima = self.data.max()
+            cs = ConfigurationSpace(seed=42)
+            for p in parameters:
+                cs.add_hyperparameter(UniformFloatHyperparameter(p[2:],
+                                      lower=minima[p] - 1, upper=maxima[p] + 1))
+        self.cs = cs
+
+    def _interpret_status(self, status):
+        if status.upper() in ["SAT", "UNSAT", "SUCCESS"]:
+            status = StatusType.SUCCESS
+        elif status.upper() in ["TIMEOUT"]:
+            status = StatusType.TIMEOUT
+        elif status.upper() in ["CRASHED"]:
+            status = StatusType.CRASHED
+        elif status.upper() in ["ABORT"]:
+            status = StatusType.ABORT
+        elif status.upper() in ["MEMOUT"]:
+            status = StatusType.MEMOUT
+        else:
+            logger.warning("Could not parse %s as a status. Valid values are: "
+                           "[SUCCESS, TIMEOUT, CRASHED, ABORT, MEMOUT]. "
+                           "Treating as CRASHED run.")
+            status = StatusType.CRASHED
+        return status
+
+    def _complete_configs(self):
+        """Either config is None or parameters an empty list.
+        After completion, every unique configuration in the data will have a
+        corresponding id. If specified via parameters, they will be replaced by
+        the config-id.
+
+        Parameters:
+        -----------
+        config_id: bool
+            if not False, column_id-column present,
+            read in configurations from somewhere
+        """
+        self.config_to_id = {}
+        parameters = [p for p in self.data.columns if p.startswith('p_')]
+        if 'config_id' in self.data.columns and len(parameters) > 0:
+            raise ValueError("Define configs either via \"p_\" or \"config\" in header, not both.")
+        elif 'config_id' in self.data.columns:
```

Comment:
I'm not sure if we want to support 'config_id' as input format. See one of my later comments in the test input file.

Reply:
I think we have to, for SMAC2. So we might as well do it for CSV...

Reply:
Yes you are right. I completely misinterpreted how the config_id thing works. After your explanation I see that it is a valid way of handling the runhistory data